src/data_insertion.py

```diff
@@ -0,0 +1,526 @@
+                        row.get("bid", None),
+                        row.get("ask", None),
+                        row.get("volume", None),
+                        row.get("openInterest", None),
+                        row.get("impliedVolatility", None),
+                        datetime.now(),
+                    ),
+                )
+
+        conn.commit()
+        conn.close()
+        print(f"Options data for {ticker_symbol} saved to database.")
+    except Exception as e:
+        print(f"Error saving options data for {ticker_symbol}: {e}")
+        raise
+
+
+@retry_on_rate_limit()
+def save_recent_news(ticker_symbol, db_name="stock_data.db"):
+    """Save only recent news (last 7 days) to the database."""
+    try:
+        ticker = yf.Ticker(ticker_symbol)
+        news = ticker.news
+
+        if not news:
+            print(f"No news data available for {ticker_symbol}")
+            return
+
+        # Filter to only include news from the last 7 days
+        recent_cutoff = datetime.now() - timedelta(days=7)
+        recent_news = [
+            item
+            for item in news
+            if datetime.fromtimestamp(item.get("providerPublishTime", 0))
+            >= recent_cutoff
+        ]
+
+        if not recent_news:
+            print(f"No recent news available for {ticker_symbol}")
+            return
+
+        conn = sqlite3.connect(db_name)
+        cursor = conn.cursor()
+
+        for item in recent_news:
+            cursor.execute(
+                """
+            INSERT OR IGNORE INTO recent_news 
+            (ticker, title, publish_date)
+            VALUES (?, ?, ?)
+            """,
+                (
+                    ticker_symbol,
+                    item.get("title", None),
+                    datetime.fromtimestamp(item.get("providerPublishTime", 0)),
+                ),
+            )
+
+        conn.commit()
+        conn.close()
+        print(f"Recent news for {ticker_symbol} saved to database.")
+    except Exception as e:
+        print(f"Error saving recent news for {ticker_symbol}: {e}")
+        raise
+
+
+def save_all_data_for_ticker(ticker_symbol, db_name="stock_data.db"):
+    """Save all essential trading data for a given ticker to the database."""
+    print(f"Saving essential trading data for {ticker_symbol}...")
+    try:
+        save_ticker_info(ticker_symbol, db_name)
+        save_historical_data(ticker_symbol, period="2y", db_name=db_name)
+        save_financial_metrics(ticker_symbol, db_name)
+        save_options_data(ticker_symbol, db_name)
+        save_recent_news(ticker_symbol, db_name)
+        print(
+            f"All essential trading data for {ticker_symbol} has been saved to {db_name}."
+        )
+        return True
+    except Exception as e:
+        print(f"Failed to complete data collection for {ticker_symbol}: {e}")
+        return False
+
+
+if __name__ == "__main__":
+    # Read tickers from nyse_tickers.txt file
+    try:
```

Comment:
This is the not the best way to get input arguments for a console application. Look at argparse or click. I prefer click.